### DISCUSSION TITLE
Annotations: Add a field to let plugins set the color of annotations

### DISCUSSION THREAD

**Discussion by lfroment0** [grafana/grafana#38370] [Dashboards, Panels & Visualizations]
**What would you like to be added**:
When using a plugin to create annotations, it would be nice to let the plugin choose the color the annotation will be drawn with.

**Why is this needed**:
This would enable plugins to embed some logic to visually differentiate annotations.

**Comment by xaviermerino** [comment]
@peterholmberg is there a way to change the color programmatically through the Annotations API?

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by mrpieper** [comment]
This would be a great feature. As we are using annotations to track Release pipelines from Azure Devops.   All of the lower environments are on same dashboard  Development, QA, Staging.     We send an annotation via api when new application code is deployed.  This allows us to track if new code caused increase in resource usage: CPU, memory, etc.     Currently Dev, QA, Staging all show with same color.  IT would be fantastic to have a way in the grafana API to specify the color, So we can have color code per enviroment.   Dev: Red, QA: Blue, Staging: Green,  etc.

**Comment by pcoriola1** [comment]
It would be really useful to to be able to use custom colors for different annotations, we have multiple deployments on the same graphs and we need to differentiate between them so would be great if we can have a way to customize that in Grafana.  Would love to have that feature.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
